src/js/game/Entities/Agent.js

```diff
@@ -19,6 +19,32 @@ module.exports = class Agent extends BaseEntity {
     }
   }
 
+  /**
+   * check whether or not the given entity can place a block
+   */
+  canPlaceBlockOver(toPlaceBlockType, onTopOfBlockType) {
+    let result = {canPlace: false, plane: ""};
+    if (onTopOfBlockType === "water" || onTopOfBlockType === "lava") {
+      if (!toPlaceBlockType.startsWith("redstoneWire") && !toPlaceBlockType.startsWith("piston") && !toPlaceBlockType.startsWith("rails")) {
+        result.canPlace = true;
+        result.plane = "groundPlane";
+      }
+    } else {
+      if (toPlaceBlockType.startsWith("redstoneWire") || toPlaceBlockType.startsWith("rails")) {
+        result.canPlace = true;
+        result.plane = "actionPlane";
+      }
+    }
+    return result;
+  }
+
+  /**
+   * player walkable stuff
+   */
+  walkableCheck() {
+    //do nothing
+  }
```

Comment:
this should be defined on `BaseEntity`

Reply:
Fair enough. I was only thinking of the Agent since no other entity places blocks. This IS definitely safer though. So great point 👍